pkg/metallb/metallb.go

```diff
@@ -392,3 +393,101 @@ func (builder *Builder) convertToStructured(unsObject *unstructured.Unstructured
 
 	return metalLb, err
 }
+
+// WithMetallbControllerConfig defines the controller config with a helm chart table.
+func (builder *Builder) WithMetallbControllerConfig(priorityClassName, runtimeClassName, deploymentName,
+	helmChartControllerTestName, helmChartKeyExample string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if !builder.Exists() {
+		glog.V(100).Infof("failed to update metallb, object doesn't exist on cluster")
+	}
+
+	glog.V(100).Infof("Defining the metallb object with Helm Chart in namespace %s",
+		builder.Definition.Namespace,
+	)
+
```

Comment:
Validate all arguments: priorityClassName, runtimeClassName, deploymentName,helmChartControllerTestName, helmChartKeyExample

Reply:
Validated

Reply:
Looking at CRD, it seems that `priorityClassName, runtimeClassName,` are optional fields. Just wonder if is needed to enforce them.

Reply:
They are optional. They can also accept an empty string. I have removed the validation

Reply:
Not sure if that's a good idea, because an empty field and a field that's not configured are not the same.

Reply:
If there is an empty field the function will fail.

Reply:
Hi I added a verification to make sure its not an empty field